js/APIMiddleware.js

```diff
@@ -0,0 +1,42 @@
+let result;
+/**
+ * API - Description
+ *
+ * @returns {type} Description
+ */
+const API = {
+    get: async (url) => {
+        await fetch(url)
+            .then(response => response.json())
+            .then((json) => {
+                if (!json.error) {
+                    result = json;
+                } else {
+                    if (json.info == "token failed to validate") {
+                        localStorage.removeItem('token');
+                        document.location.href = "index.html";
+                    } else {
+                        console.log(json);
+                    }
+                }
+            });
+        return result;
+    },
+
+    post: async (url, contentType, data) => {
```

Comment:
POST function does not check if the token is valid or not

Reply:
fixed